src/Ormolu/Config.hs

```diff
@@ -94,6 +97,21 @@ defaultConfig =
           }
     }
 
+-- | Return all dependencies of the module. This includes both the declared
+-- dependencies of the component we are working with and all potential
+-- module re-export targets.
+overapproximatedDependencies :: Config region -> Set PackageName
+overapproximatedDependencies Config {..} =
+  Set.union cfgDependencies potentialReexportTargets
+  where
+    potentialReexportTargets =
+      Set.fromList
+        . concatMap toTargetPackages
+        $ Map.elems (unModuleReexports cfgModuleReexports)
+    toTargetPackages = concatMap $ \case
+      (Nothing, _) -> []
+      (Just x, _) -> [x]
```

Comment:
I think this should not cause any problems, because the list of dependencies doesn't matter per se without the corresponding imports. So, even if you always depend on `lens`, unless you actually import `Control.Lens` it won't have any effect. The entire business of package selection is mostly an optimization procedure in the end.